### ISSUE TITLE
Strategies for .requires_grad_(False) equivalent

### ISSUE THREAD

**Issue by TugdualKerjan** [patrick-kidger/equinox#909]
Hi ! I'm currently implementing a Equinox version of Moshi's Mimi Audio codec. They use `.requires_grad_(False)` to avoid gradients for a certain jax.Array in a convolution in a certain module init:

```python3
conv = StreamingConv1d(
    in_channels,
    out_channels,
    kernel_size=2 * stride,
    stride=stride,
    causal=causal,
    groups=groups,
    bias=False,
    pad_mode="replicate",
)
if not learnt:
    where = lambda c: c.conv.conv.weight
    # TODO how to make this static ??
    self.conv = eqx.tree_at(where, conv, jnp.ones_like(conv.conv.conv.weight) / (2*stride))
```

My two ideas are: Either remember to take these out of the PyTree with partitioning when training, or somehow set these to static. I would love to have your quick opinion on what you believe would be best, and if there is another possibility I haven't taken into account ?

Have a wonderful day

**Comment by johannahaffner** [comment]
Hi Tugdual, 

you are probably looking for [jax.lax.stop_gradient](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.stop_gradient.html), which allows you to treat the array such wrapped as a constant. (During both training and inference.)

If you want to selectively freeze a model component, filtering with a custom `filter_spec` would be preferred. (More in this [thread](https://github.com/patrick-kidger/equinox/issues/154#issuecomment-1196374090), if you are interested.)

**Comment by TugdualKerjan** [comment]
Thanks for the feedback 👍
